### ISSUE TITLE
3/10 logins are not automated?

### ISSUE THREAD

**Issue by VIERcntHOLZ** [Healix/Gw2Launcher#223]
Hey,

lovely Program! I really do enjoy it.
Setup 50 accounts, all having an own local.dat.
I have set Launch Options => Automatically play after logging in + Automatically login when remembered

However, on about 3/10 attempts after launching an instance, the "login button is not automatically pressed", if thats a good way to describe it.
in 7/10 it works perfectly smooth.
Any Ideas?

PS: Also sometimes (1 out of 15 to 20 times) the game loads up, stays black and then crashes as "no response" - also that is linked to something or just "normal"?

**Comment by Healix** [comment]
Automatic logins with a remembered login is handled by GW2 itself (-autologin). A few months ago they upgraded to a new version of CoherentUI that caused a range of problems for some people, one of which is -autologin failing, and more rarely, the remembered login can fail to load.

Use the [beta version](https://github.com/Healix/Gw2Launcher/raw/master/Gw2Launcher/bin64/Beta/Gw2Launcher.exe) for a solution - Gw2Launcher will press the button instead of GW2.

The black/white screen freeze while loading character select started happening earlier this year in one of the updates for DX11. It's more likely to happen when something else using the GPU is loading while GW2 is loading character select. The beta version may help (it'll prioritize GW2 while loading), but there's ultimately no way to fully prevent it. For best results, you have to wait for character select to appear before doing _anything_ else (GW2 should be kept focused).

For that many Local.dat files, you'll want to enable: Settings > Tools > Local.dat > Faster patching.

**Comment by VIERcntHOLZ** [comment]
Thank you for your fast and indepth answer - to switch to the beta version, can i just simply override the .exe? :)

Also will opt for the fast patching.

One last thing while we are at it: I set the instances as Top left/Top right/Bottom left/bottom right via the template, now i, for test purposes (and having to authenticate yesterday still) i have "Maximum Number of active accounts" set to 4 (as I have four corners, of course), "Delay until the main window is loaded" checked and "Limit launches and authentication requests" checked.
=> Now ive heard that ANet will throttle logins after around 25? Do you have a more reliavle number? 
Therefore I'd login 25 in the morning and 25 in the evening or do you know any specific "reset" time for the login delay?

However - only one instance starts and all the others wait until the one instance is closed again => quiet inefficient. What limits the 4 simulanous instances to run? The Limiter or the "Delay until main window loaded"? I have a beefy Machine, but am afraid of running into the login throttle - any advice? :)

Thanks again!

**Comment by Healix** [comment]
You can simply overwrite it.

I haven't gotten around to creating that many accounts to test it myself. From what I've heard, the limit is around 30 accounts and people either wait hours or change their IP to launch more. The other limiter they have increases the time it takes to login with every attempt until you stop for a minute.

The limiter delays launches based on how long it took the previous account to login. Delay until loaded will pause launches until the previous launch loads character select or exits. Both of these are probably a little broken in the release version due to the CoherentUI problems.

**Comment by VIERcntHOLZ** [comment]
Done so - thanks - just checked with 25 accounts - no logging in problems and no game crashes - the beta works reliable! :)

Two - i swear - very last questions:
Still one instance is launched after another - as its still the "first launch" for those 25 accs a prompt asked if i want to start in normal mode to apply settings which, of course, I said yes to - is that the reason why not 4 instances simultanously are able to start?

Second: I email authorized all accounts yesterday, today they are asking to be authorized again - i of course checked to remember the network. Do you know why my alt accounts now ask for the third time for an email authorization while my main account never (besides day one, three months ago when i began to play again) asks for another authorization?
Of course I do have a daily Routerreset/IP change, but appearently only my alts are affected any my main account doesnt bother - any idea?

**Comment by Healix** [comment]
Yes, normal mode is 1 at a time. GW2 can't modify Local.dat otherwise.

Your main account probably has multiple IP ranges authorized, or it was authorized back when they did larger ranges. It'll stop eventually.

**Comment by VIERcntHOLZ** [comment]
Alright - so technically I currently am "farming IP Ranges" that are authorized - and if enough are for the daily resets (got 2 Ranges on my Main), it eventually will stop, correct?

Can you define when "back when they did larger ranges" was? I just am playing actively for 2,5 months now - so not that long :D

**Comment by Healix** [comment]
IP authorization remembers the first 3 numbers of your IP and allows the last number to be anything. Originally it was the first 2 or 3 numbers depending on how many people used that range.

If you keep remembering, it will eventually stop. If your main account only has 2 ranges remembered, your alts accounts probably now have the same 2 remembered.

**Comment by VIERcntHOLZ** [comment]
Thank you for your superb support.

Do you still have plans to expand the functionality of gw2login? i.e. auto character select/login, Auto close of an instance and such?

**Comment by Healix** [comment]
Automatically loading a character is against the rules. Automatically closing the game you could do by configuring a program to run after launching the account. 

If you wanted all accounts to close after 5 seconds of loading a character for example, you could Settings > Guild Wars 2 > Launch options  > Run after launching, add program: shell commands, run after loading a character, commands:
ping 127.0.0.1 -n 5
taskkill /PID %processid%

**Comment by VIERcntHOLZ** [comment]
Will try it out - sounds like a convenient Idea.

Keystrokes are not possible via shell anyhow, right? Trying to remember my IT Days in College :D

**Comment by Healix** [comment]
There are no commands for sending input.
